Ten Elephants/Ten Elephants/Models/Meals.swift

```diff
@@ -0,0 +1,116 @@
+//
+//  Meals.swift
+//  Ten Elephants
+//
+//  Created by Дарья Домрачева on 09.12.2021.
+//
+
+import Foundation
+
+struct  Meals: Decodable {
+    enum CodingKeys: String, CodingKey {
+        case meals
+    }
+    let meals: [MealModel]
+}
+
+struct MealModel: Decodable {
+    // always present
+    let name: String               // strMeal
+    let id: String                 // idMeal
+    let thumbnailLink: String      // strMealThumb
+
+    // might be present in full model
+    let category: String?          // strCategory
+    let area: String?              // strArea
+    let instructions: String?      // strInstructions
+    let youTubeLink: String?       // strYoutube
+    let tags: [String]?            // strTags
+
+    let ingredients: [Ingredient]? // Ingredient(strIngredient, strMeasure)
+
+    enum RequiredCodingKeys: String, CodingKey, CaseIterable {
+        case strMeal,
+             idMeal,
+             strMealThumb
+    }
+
+    enum OptionalCodingKeys: String, CodingKey, CaseIterable {
+        case strCategory,
+             strArea,
+             strInstructions,
+             strYoutube,
+             strTags
+    }
+
+    init(from decoder: Decoder) throws {
+        let commonContainer = try decoder.container(keyedBy: RequiredCodingKeys.self)
+
+        // these values should always be present:
+        self.name           = try commonContainer.decode(String.self, forKey: .strMeal)
+        self.id             = try commonContainer.decode(String.self, forKey: .idMeal)
+        self.thumbnailLink  = try commonContainer.decode(String.self, forKey: .strMealThumb)
+
+        let customContainer = try decoder.container(keyedBy: OptionalCodingKeys.self)
+
+        // these values can be missing in preview model
+        self.category       = try customContainer.decodeIfPresent(String.self, forKey: .strCategory)
+        self.area           = try customContainer.decodeIfPresent(String.self, forKey: .strArea)
+        self.instructions   = try customContainer.decodeIfPresent(String.self, forKey: .strInstructions)
+        self.youTubeLink    = try customContainer.decodeIfPresent(String.self, forKey: .strYoutube)
+
+        if let nonSplitTags = try customContainer.decode(String?.self, forKey: .strTags) {
```

Comment:
Кажется вместо if-else можно написсать так
```suggestion
        self.tags = try customContainer.decodeIfPresent(String.self, forKey: .strTags).flatMap{ $0.components(separatedBy: ",") }
```